src/librustc_privacy/lib.rs

```diff
@@ -937,27 +937,40 @@ impl<'a, 'tcx, 'v> Visitor<'v> for ObsoleteVisiblePrivateTypesVisitor<'a, 'tcx>
 
 struct SearchInterfaceForPrivateItemsVisitor<'a, 'tcx: 'a> {
     tcx: &'a TyCtxt<'tcx>,
-    // Do not report an error when a private type is found
-    is_quiet: bool,
-    // Is private component found?
-    is_public: bool,
+    /// The visitor checks that each component type is at least this visible
+    required_visibility: ty::Visibility,
+    /// The visibility of the least visible component that has been visited
+    min_visibility: ty::Visibility,
```

Comment:
`is_quiet` is kind of a hack, I intended to split this visitor into two visitors (for checking `T` and `Tr` in `impl Tr for T {....}` and for everything else) and also harmonize it with some stuff in `EmbargoVisitor`...
Well, I guess I'll still do it, someday.